Check that directory (still) exists

gerhardol opened the pull request:
before renabling the the FileWatcher for GitStatusMonitor

Fixes #6812

## Proposed changes

FileSystemWatcher is deactivated when the need for an update is determined and restarted when the timer fires and GE is to start a new git-status.
When (re)activating FileSystemWatcher the control raises an exception if the directory does not exist.
This change skips starting file system watcher (and git-status) and relies on the periodic 5 minute timer retry. (Normally, if a directory is no longer available the user should know about it and refresh if necessary.) 

## Test methodology

Manual (see the issue).

----

:black_nib: I contribute this code under [The Developer Certificate of Origin](../blob/master/contributors.txt).

mstv commented on GitUI/CommandsDialogs/BrowseDialog/GitStatusMonitor.cs:
```diff
@@ -198,8 +198,16 @@ private GitStatusMonitorState CurrentStatus
+                                _workTreeWatcher.EnableRaisingEvents = true;
+                            }
+
+                            if (!Directory.Exists(_gitDirWatcher.Path))
```
I have not digged into the details. Shall this really be done only if the directory does not exist yet?

gerhardol replied:
remains of my testing....

RussKie commented:
The status monitor looks arcane to me... I've looked at the changes, they seem ok, but I don't really understand what's going on (likely I haven't spent enough time going through the code).

The fact that it is an important piece of functionality and we do not have any tests for it worries me greatly.

gerhardol commented:
Squashed and rebased. This PR should not be controversial, any follow up can be done in #6844 
Merging tomorrow if not someone else does it before me.
